File: 프로그래머스/2/49993. 스킬트리/스킬트리.py
def solution(skill, skill_trees):
    answer = 0

    for skillTree in skill_trees:
        isOk = True
        curIdx = 0
        for sk in skillTree:
            if sk == skill[curIdx]:
                if curIdx < len(skill) - 1:
                    curIdx += 1
            elif sk not in skill:
                pass
            else:
                isOk = False
                break
        if isOk:
            answer += 1

    return answer

File: 프로그래머스/2/49993. 스킬트리/스킬트리.java
class Solution {
    public int solution(String skill, String[] skill_trees) {
        int answer = 0;

        for (String skillTree : skill_trees) {
            boolean isOk = true;
            int curIdx = 0;
            for (char sk : skillTree.toCharArray()) {
                if (sk == skill.charAt(curIdx)) {
                    if (curIdx < skill.length()-1) {
                        curIdx++;
                    }
                } else if (!skill.contains(Character.toString(sk))) {
                    
                } else {
                    isOk = false;
                    break;
                }
            }
            if (isOk) {
                answer++;
            }
        }
        return answer;
    }
}